### PR TITLE
All the fixes

### LINKED ISSUE #64: Support the `--web-compiler=dartdevc` flag for use with `pub serve`
cc @matanlurey @kwalrath @kevmoo

### LINKED ISSUE #66: pub run angular_test leaves `run test` processes active after its done
This is on MacOS, using Dart 1.24.0-dev.6.8.

After a few (sparse) hours of testing, `pub run angular_test` just stopped working. I looked at my active processes and found over a dozen latent `dart run test ...` processes
```
28566 ttys009    0:01.69 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
28634 ttys009    0:01.81 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
28702 ttys009    0:01.67 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
28737 ttys009    0:01.86 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
28788 ttys009    0:01.71 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
28925 ttys009    0:01.58 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
28979 ttys009    0:02.32 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
29040 ttys009    0:01.75 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=dartium
52083 ttys009    0:01.46 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=content-shell
52169 ttys009    0:01.39 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=content-shell
53321 ttys009    0:01.46 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=content-shell
53839 ttys009    0:01.30 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=content-shell
94408 ttys009    0:02.29 /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/dart /opt/homebrew/Cellar/dart/1.24.0-dev.6.8/libexec/bin/snapshots/pub.dart.snapshot run test --pub-serve=8080 -t aot --platform=content-shell
```

Killing all these latent processes allowed `pub run angular_test` to work again.

cc @matanlurey @kwalrath @kevmoo

### LINKED ISSUE #67: Support --test-args and --pub-args...
To be able to pass any arbitrary arguments to `pub run test` or `pub serve`.

### PR DESCRIPTION
**Pull request by matanlurey**
Decided not to try and support specialized CLI arguments (Closes #67).

Major changes are detailed in the CHANGELOG.md. Overall this will be easier to maintain. Also added tests for much of the CLI options parsing, since that seems important to get right.

Closes #64.
Closes #66.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+- Option `--platform` (`-p`) is now **Deprecated**, and no longer has a default
+  value of `content-shell`, which was not always installed on host machines.
+  Instead use `--test-arg`:
+
+```bash
+$ pub run angular_test --test-arg=--platform=content-shell
+```
+
+- Option `--name` (`-n`) and `--simple-name` (`-N`) are also deprecated. Use
+  `--test-arg=--name=` and `--test-arg=--simple-name=` instead.
```
Comment (grouma) [inline CHANGELOG.md:34]: Is there documentation for this besides the changelog and the sample flags in the README? It wasn't obvious at first that you can provide multiple --test-arg's

File: lib/src/bin/runner.dart [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    log('${options.pubBin} ${options.pubArgs.join(' ')}');
+    pubServeProcess = await Process.start(options.pubBin, options.pubArgs);
+    Uri serveUri;
+    var testsRunning = false;
+
+    final stdOutDone = _toLines(pubServeProcess.stdout).forEach((line) async {
+      if (serveUri == null) {
+        final serveMatch = _serveRegExp.firstMatch(line);
+        if (serveMatch != null) {
+          serveUri = Uri.parse(serveMatch[1]);
```
Comment (grouma) [inline lib/src/bin/runner.dart:54]: Can this have an array out of bound exception?

**Review by grouma — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (matanlurey) [reply CHANGELOG.md:34]: Done.

File: lib/src/bin/runner.dart (hunk 2 above)
Comment (matanlurey) [reply lib/src/bin/runner.dart:54]: This is just copied from the original code /shrug

**Comment by grouma** [comment]
Travis is super slow, consider extending the timeout by using the test-arg flag. For the test package we have settled on 4x the default.

**Comment by grouma** [comment]
Woot!

**Pull request merged**